File: src/test/testdata/inspections/flushInsideLoop/Test3.java
package inspections.flushInsideLoop;

import com.badlogic.gdx.graphics.g2d.CpuSpriteBatch;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Matrix4;

public class Test3 {

    public Test3() {

    }

    public Test3(SpriteBatch batch) {
        batch.flush();
    }

    public Test3(int i) {
        new SpriteBatch().setTransformMatrix(new Matrix4());
    }

    public Test3(float f) {
        new CpuSpriteBatch().setTransformMatrix(new Matrix4());
    }

    public static void sMethod() {
        new SpriteBatch().flush();
    }

    public void method1() {
        new SpriteBatch().setTransformMatrix(new Matrix4());
    }

    public void method2(SpriteBatch batch) {
        batch.setTransformMatrix(null);
    }

    public void method3() { new CpuSpriteBatch().setTransformMatrix(new Matrix4());}


}

File: src/test/testdata/inspections/flushInsideLoop/Test3.kt
package flushInsideLoop

import com.badlogic.gdx.graphics.g2d.CpuSpriteBatch
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.math.MathUtils
import com.badlogic.gdx.math.Matrix4

@Suppress("UNUSED_VARIABLE")
fun t() {

  while (MathUtils.randomBoolean()) {

    fun f() {
      SpriteBatch().flush()
    }

    CpuSpriteBatch().setTransformMatrix(Matrix4())

    val c = Test3()
    val d = <warning>Test3(SpriteBatch())</warning>
    val e = <warning>Test3(1)</warning>
    val f = Test3(1.0f)

    c.<warning>method1()</warning>
    c.<warning>method2(SpriteBatch())</warning>
    c.method3()

    Test3.<warning>sMethod()</warning>


  }

}